From 63421f1c92403149be838c64ebb731778c148dde Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Thu, 7 Sep 2017 07:36:39 -0400
Subject: [PATCH 1/1] Ignore errors from the File::LibMagic describe_filename
 function.

Newer versions of amavisd-new can make use of the File::LibMagic
library (as opposed to the "file" executable) to get information about
attachments. Using the library should be faster, but when it fails, it
does so in a novel way:

  (02859-21) File::LibMagic::describe_filename failed on p003: libmagic
  JPEG image data, Exif standard: [TIFF image data, big-endian,...

This commit adds such lines to the global ignore list, since they
don't provide us with any useful information.
---
 amavis-logwatch | 1 +
 1 file changed, 1 insertion(+)

diff --git a/amavis-logwatch b/amavis-logwatch
index 448de3a..ac93cbc 100644
--- a/amavis-logwatch
+++ b/amavis-logwatch
@@ -2033,6 +2033,7 @@ sub create_ignore_list() {
 
    push @ignore_list_final, qr/^Inserting header field: X-Amavis-Hold: /;
    push @ignore_list_final, qr/^Decoding of .* failed, leaving it unpacked: /;
+   push @ignore_list_final, qr/^File::LibMagic::describe_filename failed on p\d+: /;
 
       # various forms of "Using ..."
       # more specific, interesting variants already captured: search "Using"
-- 
2.13.0

